### ISSUE TITLE
[bug] "unknown module referenced" errors when re-init StreamAlert

### ISSUE THREAD

**Issue by chunyong-lin** [airbnb/streamalert#1047]
## Background
During a new deployment of StreamAlert based on branch `release-3-0-0` to stage environment, re-run `python manage.py init` if first init was failed and it would be interrupted with error messages
```
Error: module 'metric_filters_Classifier_FailedParses_PROD_AGGREGATE': unknown module referenced: classifier_prod_lambda

Error: module 'metric_filters_Classifier_SQSFailedRecords_PROD_AGGREGATE': unknown module referenced: classifier_prod_lambda

Error: module 'metric_filters_Classifier_FirehoseRecordsSent_PROD_AGGREGATE': unknown module referenced: classifier_prod_lambda
(...)
```

### Description
My investigation is that terraform configuration file `terraform/prod.tf.json` for `prod` cluster (in my case) is deleted but it has been saved in terraform state file locally. So terraform complaint the reference is not found.

### Steps to Reproduce
```
python manage.py configure aws_account_id 11112222
python manage.py configure prefix MY_PREFIX
python manage.py init
[If bad luck, the first init failed or force trigger the init to fail using our imagination]
python manage.py init
```
We will see the error messages mentioned above.

## Desired Change
Don't delete terraform configuration files for clusters when re-run `python manage.py terraform init`.
One potential fix is to set `init` to `False` in the handler of `TerraformDestroyCommand` class.
https://github.com/airbnb/streamalert/blob/b9a1b6c642ce320525af73abc33680b67e0535ce/streamalert_cli/terraform/handlers.py#L227

**Comment by ryandeivert** [comment]
fixed with #1060

**Comment by chunyong-lin** [comment]
Reopen this issue because the fix in #1060 is not ideal. It fixed `unknown module referenced`, however, it  introduce the bug filed in issue #1048 when we want to destroy StreamAlert.

In PR #1060, it changed the `init` flag from `True` to `False` to fix `unknown module referenced`. However, it requires `init` flag set to `True` to change the terraform backend to local, in order to prevent the s3 bucket that stores terraform remote state file getting deleted.
https://github.com/airbnb/streamalert/blob/7d198a3273781f66465420e90886a3ce53ec7559/stream_alert_cli/terraform/generate.py#L149-L157

I do not have solution now.

**Comment by jack1902** [comment]
I'll have another look at this today, since the backend needs to change to local

**Comment by chunyong-lin** [comment]
Fixed in PR #1093
